### ISSUE TITLE
About the size of the search area during testing

### ISSUE THREAD

**Issue by TCBocean** [MegviiDetection/video_analyst#126]
I noticed that during training, the search area calculation method of x frame and z frame is similar to siamfc, but when testing, they appear different. This difference comes from the get_crop function (https://github.com/MegviiDetection/video_analyst/blob/b786f48e92d6cd3d2759d55372e8c8ef76f6b5e5/videoanalyst/pipeline/utils/crop.py#L108).
In the z frame during the test, since the x_size parameter is not passed to the get_crop function, the calculation of the search area of the z frame is similar to siamfc.
https://github.com/MegviiDetection/video_analyst/blob/b786f48e92d6cd3d2759d55372e8c8ef76f6b5e5/videoanalyst/pipeline/tracker_impl/siamfcpp_lt_track.py#L155
In the x frame during the test, since the x_size parameter is passed to the get_crop function, the search area of the x frame is calculated as x_size/(z_size / s_crop).
https://github.com/MegviiDetection/video_analyst/blob/b786f48e92d6cd3d2759d55372e8c8ef76f6b5e5/videoanalyst/pipeline/tracker_impl/siamfcpp_lt_track.py#L235

Then there is a problem: 
During training, the size of the search area of the x and z frames is only related to the target size, and not related to the size of x_size and z_size. 
In the test, the size of the search area of the x and z frames is not only related to the target size, but also related to the size of x_size and z_size. This causes the search area of the x frame to be much larger than the z frame.

I think more contextual information will affect the performance of the tracker.
 So I want to ask, does increasing the search area affect the performance of the tracker during testing? Or what misunderstanding I have, I hope you can correct my misunderstanding.
Thanks~

**Comment by lzx1413** [comment]
@TCBocean  In fact, the crop size of target is only related with target size we defined (which is same with z_size). And for the test frame, we can change the x_size to enlarge or reduce the search area, which is a hyper-parameter for objects with different move speed.

**Comment by TCBocean** [comment]
@lzx1413 
Thank you for your answer, but the content of the answer does not seem to match my question.
But I went to read the data preprocessing part of the code again, and I found that I had overlooked a detail, which is the following line of code:
https://github.com/MegviiDetection/video_analyst/blob/01b7ad278b828a3f7ff7a0488c5ca8f055240192/videoanalyst/data/utils/crop_track_pair.py#L201
During training, the search area of frame z is much larger than that of frame x, so the same operation is done during testing.At first I thought that the calculation method of the search area of the x frame and the search area of the z frame is only related to the target size
I want to ask another question in this question. I don't know if this is allowed. If it doesn't work, I can open another question.
 Regarding the loss function of ctr, he has no problem, but I did not understand it. It is not like the BCELoss I know. Is there the source of this loss function?

**Comment by lzx1413** [comment]
@TCBocean the ctr loss is defined [here](https://github.com/MegviiDetection/video_analyst/blob/master/videoanalyst/model/loss/loss_impl/sigmoid_ce_centerness.py). It is one BCE loss actually. It just one implementation with numerically stable formulate. You can replace it with the standard BCELoss defined in pytorch.

**Comment by TCBocean** [comment]
Thank you for your answer
